crates/sc-proof-of-time/src/lib.rs

```diff
@@ -2,102 +2,115 @@
-
-pub use state_manager::{
-    PotConsensusState, PotGetBlockProofsError, PotStateSummary, PotVerifyBlockProofsError,
-};
-pub use time_keeper::TimeKeeper;
-
-// TODO: change the fields that can't be zero to NonZero types.
-// TODO: CLean up unused fields
-#[derive(Debug, Clone)]
-pub struct PotConfig {
-    /// PoT seed used initially when PoT chain starts.
-    pub initial_seed: PotSeed,
-
-    /// PoT key used initially when PoT chain starts.
-    pub initial_key: PotKey,
-
-    /// Frequency of entropy injection from consensus.
-    pub randomness_update_interval_blocks: BlockNumber,
-
-    /// Starting point for entropy injection from consensus.
-    pub injection_depth_blocks: BlockNumber,
-
-    /// Number of slots it takes for updated global randomness to
-    /// take effect.
-    pub global_randomness_reveal_lag_slots: SlotNumber,
-
-    /// Number of slots it takes for injected randomness to
-    /// take effect.
-    pub pot_injection_lag_slots: SlotNumber,
-
-    /// If the received proof is more than max_future_slots into the
-    /// future from the current tip's slot, reject it.
-    pub max_future_slots: SlotNumber,
-
-    /// Total iterations per proof.
-    pub pot_iterations: NonZeroU32,
+use subspace_core_primitives::PotCheckpoints;
+#[cfg(feature = "pot")]
+use tracing::error;
+use tracing::{debug, trace};
+
+pub trait PotSlotWorker<Block>
+where
+    Block: BlockT,
+{
+    /// Called when new proof of time is available for slot.
+    ///
+    /// NOTE: Can be called more than once in case of reorgs to override old slots.
+    fn on_proof(&mut self, slot: Slot, checkpoints: PotCheckpoints);
 }
 
-/// Components initialized during the new_partial() phase of set up.
-#[derive(Debug)]
-pub struct PotComponents {
-    /// PoT seed used initially when PoT chain starts.
-    // TODO: Remove this from here, shouldn't be necessary eventually
-    pub(crate) initial_seed: PotSeed,
-
-    /// PoT key used initially when PoT chain starts.
-    // TODO: Remove this from here, shouldn't be necessary eventually
-    pub(crate) initial_key: PotKey,
+/// Start a new slot worker.
+///
+/// Every time a new slot is triggered, `worker.on_slot` is called and the future it returns is
+/// polled until completion, unless we are major syncing.
+pub async fn start_slot_worker<Block, Client, SC, Worker, SO, CIDP>(
+    slot_duration: SlotDuration,
+    #[cfg(feature = "pot")] client: Arc<Client>,
+    select_chain: SC,
+    worker: Worker,
+    sync_oracle: SO,
+    create_inherent_data_providers: CIDP,
+    mut slot_info_stream: PotSlotInfoStream,
+) where
+    Block: BlockT,
+    Client: ProvideRuntimeApi<Block> + HeaderBackend<Block>,
+    Client::Api: SubspaceRuntimeApi<Block, FarmerPublicKey>,
+    SC: SelectChain<Block>,
+    Worker: PotSlotWorker<Block> + SimpleSlotWorker<Block> + Send + Sync,
+    SO: SyncOracle + Send,
+    CIDP: CreateInherentDataProviders<Block, ()> + Send + 'static,
+{
+    #[cfg(feature = "pot")]
```

Comment:
I wanted to avoid `let block_authoring_delay = {}` here. It isn't nice either way until we remove feature flag, also non-pot version below is just a workaround for it to compile, it is not actually used anywhere, it just happens to be the case that this crate is compiled with the rest during clippy lints and tests even when PoT is not enabled otherwise.